common/src/test/scala/uk/ac/wellcome/utils/TryBackoffTest.scala

```diff
@@ -58,6 +64,23 @@ class TryBackoffTest
     f.calls.length shouldBe finalLength
   }
 
+  it("should eventually call terminalFailureHook for a a function that always fails") {
+    val f = new TryBackoffHelper()
+
+    tryBackoff.wasCalled = false
+    tryBackoff.run(f.alwaysFails, system)
+
+    Thread.sleep(10000)
+    val finalLength = f.calls.length
+    Thread.sleep(5000)
+
+    f.calls.length shouldBe finalLength
```

Comment:
As discussed, let’s cut this middle bit – so this test only fails if the terminal failure hook is never called.